[ES6] Breaking: avoid string concatenation

niedzielski opened the pull request:
Favor template literals and avoid unnecessary string concatenation.
Also update test/test.js with the new rules and no other changes.

niedzielski committed:
[ES6] Breaking: avoid string concatenation

Favor template literals and avoid unnecessary string concatenation.
Also update test/test.js with the new rules and no other changes.

niedzielski committed:
[ES6] Breaking: avoid string concatenation

Favor template literals and avoid unnecessary string concatenation.
Also update test/test.js with the new rules and no other changes.

Krinkle commented on test/fixtures/server/valid.js:
```diff
@@ -17,4 +17,11 @@
+
+	// Rule: prefer-template
+	// eslint-disable-next-line no-unused-expressions
+	`a${global}`;
```
`{var}` or `{ var }`? I've always seen and used `{var}`, but the next file in the this uses spaced `{ var }`.

niedzielski replied:
Fixed. Added template-curly-spacing.

niedzielski committed:
[ES6] Breaking: style template literal variables

Enable template-curly-spacing which enforces a consistent variable
reference style in template literals.

https://eslint.org/docs/rules/template-curly-spacing

niedzielski committed:
[ES6] Breaking: style template literal variables

Enable template-curly-spacing which enforces a consistent variable
reference style in template literals.

https://eslint.org/docs/rules/template-curly-spacing

jdforrester commented on language/es6.json:
```diff
@@ -13,6 +13,7 @@
     "no-useless-computed-key": "error",
     "no-useless-concat": "error",
     "prefer-template": "error",
+    "template-curly-spacing": "error",
```
This is a "positive" rule (not starting with `no-`) that triggers a test failure if used wrongly and should be covered in the positiveFailures files, except we don't test language rules yet. Meh.

jdforrester committed:
Provide ability to lint JSON files (#167)

* Upgrade js-yaml sub-dependency from 3.12.1 to 3.13.0 for DoS fix

* Update eslint to 5.16

Only new rule is prefer-named-capture-group, which is a ECMAScript 2018
feature, so we can ignore it.

* Provide ability to lint JSON files (Fixes #166)

Users will have to adjust the src match from *.js to also cover JSON
files, but there's no need to set the `extensions` options value as
well.

* test: Run eslint for json files too

* build: Update js-yaml locked dependency for audit

jdforrester committed:
Tag v0.12.0 (#170)

hknustwmf committed:
Added prefer-const rule to es6.json in eslint-config-wikimedia (#171)

Bug: T224509

Pchelolo committed:
node: Add no-buffer-constructor rule

The buffer constructor is deprecated in Node.js and is causing
a warning in Node 10. As we're switching more services to Node 10,
we need to lint for buffer constructors in the codebase.

Closes #172.

edg2s committed:
Enable no-jquery/no-sizzle rule (#175)

Fixes #174

Remove no-jquery/no-size test as it is now part
of deprecated-1.8

jdforrester committed:
jquery: Enable new rule, no-class-state (#176)

jdforrester committed:
Merge branch 'breaking/string-concat' of https://github.com/wikimedia/eslint-config-wikimedia into breaking/string-concat
